app/src/main/java/com/xa/xpensauditor/Day_wise_visualization.java

```diff
@@ -0,0 +1,204 @@
+                    int dayyy;
+
+                    @Override
+                    public void onDataChange(@NonNull DataSnapshot datasnapshot) {
+
+                        for(DataSnapshot snapshot :datasnapshot.getChildren() ){
+                            String amm=snapshot.child("Amount").getValue().toString();
+                            System.out.println(" Finallyyy   "+amm);
+                            int am1=Integer.valueOf(amm);
+                            tot=am1+tot;
+      dumm=snapshot.child("Day").getValue().toString();
+      dumm1=Integer.valueOf(dumm);
+      System.out.println("dummmmm   "+dumm1+"    "+dayyy);
+      if(dayyy==dumm1){
+          System.out.println("inside");
+          int temp=0;
+          temp=hm.get(dayyy);
+      temp=temp+am1;
+          System.out.println("dummmmm1   "+dayyy+"    "+temp);
+      hm.put(dayyy,temp);
+      }
+      else{
+      dayy=snapshot.child("Day").getValue().toString();
+      dayyy=Integer.parseInt(dayy);
+     hm.put(dayyy,am1);}
+                 }
+
+                        ArrayList<Integer> sortedKeys
+                                = new ArrayList<Integer>(hm.keySet());
+
+                        Collections.sort(sortedKeys);
+
+                        // Display the TreeMap which is naturally sorted
+                        for (Integer x : sortedKeys) {
+                            data.add(new ValueDataEntry(x,hm.get(x)));
+                        }
+hm.clear();
+                        Column column = cartesian.column(data);
+                        column.tooltip()
+                                .titleFormat("{%X}")
+                                .position(Position.CENTER_BOTTOM)
+                                .anchor(Anchor.CENTER_BOTTOM)
+                                .offsetX(0d)
+                                .offsetY(5d)
+                                .format("${%Value}{groupsSeparator: }");
+
+                        cartesian.animation(true);
+                        cartesian.title("Expenses day wise");
+
+                        cartesian.yScale().minimum(0d);
+
+                        cartesian.yAxis(0).labels().format("${%Value}{groupsSeparator: }");
+
+                        cartesian.tooltip().positionMode(TooltipPositionMode.POINT);
+                        cartesian.interactivity().hoverMode(HoverMode.BY_X);
+
+                        cartesian.xAxis(0).title("Dates");
+                        cartesian.yAxis(0).title("Expences");
+
+                        anyChartView.setChart(cartesian);
+                        // System.out.println(Arrays.toString(arr.toArray())+" arrrrrayyyyy  ");
+                    }
+
+                    @Override
+                    public void onCancelled(@NonNull DatabaseError error) {
+
+                    }
+                });
+                System.out.println(Arrays.toString(arr.toArray())+" arrrrrayyyyy2  ");
+            }
+            System.out.println(data+"   data2222  ");
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAroundCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>WhitespaceAround: '+' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3)